sqlmesh/core/context_diff.py

```diff
@@ -41,9 +41,9 @@ class ContextDiff(PydanticModel):
     """Whether the currently stored environment record is in unfinalized state."""
     create_from: str
     """The name of the environment the target environment will be created from if new."""
-    added: t.Set[str]
+    added: t.Dict[str, Snapshot]
     """New models."""
-    removed: t.Set[str]
+    removed: t.Dict[str, Snapshot]
```

Comment:
Ditto

Reply:
If I'm reading the code correctly, removed snapshots are not part of ContextDiff. If these were added to ContextDiff, then yes, the dicts could be sets.

Reply:
I see. Do we actually need a full-blown snapshot or the Table Info from the environment will suffice?

Reply:
The TableInfo would suffice. We have the removed Snapshot in a local var when constructing context diff. I can add a removed_snapshots dict if that's preferable. What do you think?